brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/spec/VanillaSoftwareProcessLifecycleModifier.java

```diff
@@ -65,96 +74,94 @@ public void apply(EntitySpec<?> entitySpec, NodeTemplate nodeTemplate, Topology
 
         // Applying operations
         final Map<String, Operation> operations = getStandardInterfaceOperations(nodeTemplate, topology);
-        if (!operations.isEmpty()) {
-            applyLifecycle(operations, ToscaNodeLifecycleConstants.CREATE, entitySpec, VanillaSoftwareProcess.INSTALL_COMMAND, nodeTemplate, topology);
-            applyLifecycle(operations, ToscaNodeLifecycleConstants.CONFIGURE, entitySpec, VanillaSoftwareProcess.CUSTOMIZE_COMMAND, nodeTemplate, topology);
-            applyLifecycle(operations, ToscaNodeLifecycleConstants.START, entitySpec, VanillaSoftwareProcess.LAUNCH_COMMAND, nodeTemplate, topology);
-            applyLifecycle(operations, ToscaNodeLifecycleConstants.STOP, entitySpec, VanillaSoftwareProcess.STOP_COMMAND, nodeTemplate, topology);
-            if (!operations.isEmpty()) {
+        for (String opKey : operations.keySet()) {
+            if (!lifeCycleMapping.containsKey(opKey)) {
                 LOG.warn("Could not translate some operations for " + nodeTemplate + ": " + operations.keySet());
+                continue;
             }
+
+            Operation op = operations.get(opKey);
+            applyLifecycle(op, entitySpec, lifeCycleMapping.get(opKey), nodeTemplate, topology);
         }
     }
 
     protected Map<String, Operation> getStandardInterfaceOperations(NodeTemplate nodeTemplate, Topology topology) {
         Map<String, Operation> operations = MutableMap.of();
-        Interface indexedNodeTemplateInterface = findInterfaceOfNodeTemplate(
-                getIndexedNodeTemplate(nodeTemplate, topology).get().getInterfaces(), VALID_INTERFACE_NAMES);
-        Interface nodeTemplateInterface = findInterfaceOfNodeTemplate(
+        IndexedArtifactToscaElement indexedNodeTemplate = getIndexedNodeTemplate(nodeTemplate, topology).get();
+
+        Interface indexedNodeTemplateInterface = NodeTemplates.findInterfaceOfNodeTemplate(
+                indexedNodeTemplate.getInterfaces(), VALID_INTERFACE_NAMES);
+
+        Interface nodeTemplateInterface = NodeTemplates.findInterfaceOfNodeTemplate(
```

Comment:
Think it would be cleaner if you changed `findInterfaceOfNodeTemplate` to return `Optional<Interface>` rather than `null` when no such interface exists.